aa-sdk/core/src/middleware/defaults/7702gasEstimator.ts

```diff
@@ -0,0 +1,44 @@
+import { AccountNotFoundError } from "../../errors/account.js";
+import type { UserOperationStruct } from "../../types.js";
+import type { ClientMiddlewareFn } from "../types";
+import { defaultGasEstimator } from "./gasEstimator.js";
+
+/**
+ * A middleware function to estimate the gas usage of a user operation with an optional custom gas estimator.
+ * This function is only compatible with accounts using EntryPoint v0.7.0.
+ *
```

Comment:
we should add an `@example` block here as well and it should look like:
````
@example
```ts twoslash
....
```
````

Reply:
the `twoslash` piece will ensure that the code is valid when the docs site is built AND add the on hover interaction to the docs